apps/frontend/src/pages/CreateListing.tsx

```diff
@@ -1,5 +1,281 @@
+    try {
+      navigator.geolocation.getCurrentPosition((position) => {
+        newListingObject.listing.location.longitude = position.coords.longitude;
+        newListingObject.listing.location.latitude = position.coords.latitude;
+      });
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }, [newListingObject.listing.location]);
+
+  // Updates and sends the newListingObject, to the server via post under /api/listing
+  const sendPostToCreateListing = async () => {
+    // In order to make sure that the images are retrieved before submitting
+    const successImages = await uploadImages();
+    let successLocation: boolean = true;
+    if (newListingObject.listing.location.latitude !== 0) {
+      successLocation = false;
+      getUserLocation().then(() => {
+        successLocation = true;
+      });
+    }
+    //console.log("Image Successful", imageUploadSuccessful)
+    console.log("New Listing Object", newListingObject);
+    if (successImages && successLocation) {
+      _axios_instance
+        .post("/listing", newListingObject)
+        .then((response) => {
+          alert("Listing Created!");
+          console.log("Response to upload", response);
+        })
+        .catch((error) => {
+          console.log("Response to error", error);
+          alert("Listing Creation Failed");
+        });
+    } else if (!successImages) {
+      alert("Images failed to upload, please try again later");
+    } else {
+      // Currently this is added to catch if the location is not set, we could default this to the location of the university instead
+      alert(
+        "Error occurred when creating a listing, you may need to enable location permissions for this site",
+      );
+    }
+  };
+
+  // This function makes sure that the passed in url is a base64 data string
+  const isImageValid = (url: string) => {
+    try {
+      return url.startsWith("data");
+    } catch (error) {
+      return error === TypeError && url === undefined;
+    }
+  };
+
+  // Request the user's location on load
+  useEffect(() => {
+    getUserLocation().then((r) => console.log(r));
+  }, [getUserLocation]);
+
+  // Uploads the images to the s3 server, this is handled seperately
+  // This may need to be modified to use File objects instead of the file strings
+  const uploadImages = async () => {
+    try {
+      await Promise.all(
+        images.map(async (image) => {
+          try {
+            const response = await _axios_instance.post(
+              "/images",
+              { image: image },
+              {
+                headers: {
+                  "Content-Type": "multipart/form-data",
+                },
+              },
+            );
+            newListingObject.listing.images.push({ url: response.data.url });
+          } catch (error) {
+            console.error("Error uploading image:", error);
+          }
+        }),
+      );
+      return true;
```

Comment:
I believe with the way this is set up, any error uploading a specific image would be logged to the console, but the return would still be true. The outside catch would never see an error so the function would never return false, even if no image uploaded successfully.

Reply:
Sounds good, I'll work on that today, there may be a couple things like that as I'm often working on this stuff quite late

Reply:
Hey Scott, if you could take another look at this code snippet, that'd be much appreciated!

Reply:
I'm still not 100% sure this works.

`try {`
`  for each image {`
`    try {`
`       (error thrown here on image upload failure)`
`    } catch {`
`       (error caught here)`
`    }`
`    // no error at this point since it was caught above`
`  }`
`} catch {`
`    // never triggered because error was already caught`
`}`

I would probably just have a single try/catch wrapping all the image uploads, since it is pretty unlikely that some would succeed and others would fail. If you want each upload to have its own catch and the promise to have its own as well, I think you can do a .catch() on the promise, or manually throw an error

https://stackoverflow.com/questions/24977516/catching-errors-in-javascript-promises-with-a-first-level-try-catch

Reply:
I think can see your point, but I think my brain is slowly dissolving...

Would this not work to make sure that all of the images have been uploaded if the user has a faulty connection with a high packet loss? I know it'd be cleaner to have the try catches attached to the axios promise, but I was trying that before and for whatever reason that weird test has started passing with this new implementation